engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/serialization/KafkaAvroDeserializationSchemaFactory.scala

```diff
@@ -103,26 +65,38 @@ abstract class KafkaAvroKeyValueDeserializationSchemaFactory
       @transient
       private lazy val valueDeserializer = createValueDeserializer[V](valueSchemaDataOpt, kafkaConfig)
 
-      override def deserialize(consumerRecord: ConsumerRecord[Array[Byte], Array[Byte]]): O = {
-        val key = keyDeserializer.deserialize(consumerRecord.topic(), consumerRecord.key())
-        val value = valueDeserializer.deserialize(consumerRecord.topic(), consumerRecord.value())
-        val obj = createObject[K, V](key, value, consumerRecord.topic())
-        obj.asInstanceOf[O]
+      override def deserialize(record: ConsumerRecord[Array[Byte], Array[Byte]]): ConsumerRecord[K, V] = {
+        val key = keyDeserializer.deserialize(record.topic(), record.key())
+        val value = valueDeserializer.deserialize(record.topic(), record.value())
+        new ConsumerRecord[K, V](
+          record.topic(),
+          record.partition(),
+          record.offset(),
+          record.timestamp(),
+          record.timestampType(),
+          ConsumerRecord.NULL_CHECKSUM.toLong, // ignore deprecated checksum
+          record.serializedKeySize(),
+          record.serializedValueSize(),
+          key,
+          value,
+          record.headers(),
+          record.leaderEpoch()
+        )
       }
 
-      override def isEndOfStream(nextElement: O): Boolean = false
+      override def isEndOfStream(nextElement: ConsumerRecord[K, V]): Boolean = false
+
+      override def getProducedType: TypeInformation[ConsumerRecord[K, V]] = {
+        if (kafkaConfig.useStringForKey) {
+          val clazz = classTag[ConsumerRecord[String, V]].runtimeClass.asInstanceOf[Class[ConsumerRecord[K, V]]]
```

Comment:
1. Can't you just do `classOf[ConsumerRecord[String, V]]`? without classTag? Also IMO there is no need to have this if-else here because this class object is always the same because generic type erassure.
2. Can you left TODO that it would be nice to build `TypeInformation` for `ConsumerRecord` using `createKeyTypeInfo` and `createValueTypeInfo`?

Reply:
Fixed and added some TODO explanation.